internal/service/vmservice/vm.go

```diff
@@ -346,7 +346,8 @@ func getMachineAddresses(scope *scope.MachineScope) ([]clusterv1.MachineAddress,
 	return addresses, nil
 }
 
-func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneResponse, error) {
+// TODO: split function to avoid gocyclo.
+func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneResponse, error) { //nolint: gocyclo
```

Comment:
Nope, please don't turn off the linter.

Reply:
I'm not reviewing the rest of this func as my earlier comments request extensive changes here.

Reply:
ok, my local new tests do not go into gocyclo.